### ISSUE TITLE
Repair error

### ISSUE THREAD

**Issue by s3kShUn61** [JoSchaap/GoT_Wasteland_V2.Stratis#101]
I attempted to repair an APC that had red damaged wheels. When I tried to use the player inventory menu repair kit I got this error and was unable to repair. http://i.imgur.com/ESo2Ul9.jpg

**Comment by Nebular-DH** [comment]
Occurs because the vehicle has no glasses.

**Comment by AgentRev** [comment]
The real problem isn't that the vehicle has no glasses, but that the guy who made the repair script was too stupid to dynamically acquire hitpoints for the vehicle instead of relying on static values.

**Comment by JoSchaap** [comment]
yea it seems since ToH there is a gethitpointdamage function :)

```
   (vehicle player) getHitPointDamage "hitEngine";
```

and it shouldnt be too hard to grab the hotpoints from cfgvehicles :)

```
   count(configfile >> "CfgVehicles" >> "B_MRAP_01_gmg_F" >> "HitPoints"");
```

**Note to self: http://community.bistudio.com/wiki/getHitPointDamage

**Comment by AgentRev** [comment]
You don't have to set each hitpoint individually, a simple `vehicle setDamage 0` will repair all hitpoints.

**Comment by JoSchaap** [comment]
then i really dont get it why they added the hitpoints at all..

maybe they tried to fix fuel leaks in arma2 this way or something :/

**Comment by AgentRev** [comment]
Probably because having a simple blown tire or broken window doesn't count as the vehicle being broken, but that the repair kit should fix it nonetheless.

**Comment by AgentRev** [comment]
You should modify `repair.sqf` to something like this:

```
if (damage _currVehicle > 0.01) then
{
    ...
    _totalDuration = ceil ((damage _currVehicle) * 10);
    ...
}
else
{
    _currVehicle setDamage 0;
    hint "Vehicle repaired.";
};
```

**Comment by JoSchaap** [comment]
That would suffice but will need some testing. 

Atm my phone screen (well dimmed) is all i can handle. Gotta love migraine.. 

All the input is Appreciated and as soon as the headackes cool down ill get crackin on it :)

**Comment by JoSchaap** [comment]
aa5967fa0f6b23eba62b209c38d546b9601d70b7
e5cef0a2b001e0555893b6f78b0524fc8c74f230
afae17d8fc0b369d7233fc03bfc8b1bc5b1ec926
b68bf16fc950b5419528ea2ea50788397705abfe

fixed in v2.4
